chemprop/utils.py

```diff
@@ -359,10 +359,11 @@ def create_logger(name: str, save_dir: str = None, quiet: bool = False) -> loggi
     :param quiet: Whether the stream handler should be quiet (i.e., print only important info).
     :return: The logger.
     """
-    logger = logging.getLogger(name)
 
-    if logging.getLogger().hasHandlers():
-        return logger
+    if name in logging.root.manager.loggerDict:
+        return logging.getLogger(name)
+    
+    logger = logging.getLogger(name)
```

Comment:
I think it would be more elegant to first assign `logger` and then either return it or walk through the rest of the function depending on `logging.root.manager.loggerDict`, but that is maybe just a question of taste. I.e.

```
logger = logging.getLogger(name)
if name in logging.root.manager.loggerDict:
    return logger
```

instead of

```
if name in logging.root.manager.loggerDict:
    return logging.getLogger(name)
logger = logging.getLogger(name)
```

Reply:
The problem with the `getLogger` function is that it will retrieve a logger if it exists already or create one without indicating which it did.  By assigning `logger` first, you cause it to be written into `logging.root.manager.loggerDict` and you would always have the check be true.

Reply:
My bad, you are right!